include/accessor.h

```diff
@@ -128,19 +180,19 @@ class accessor<DataT, Dims, Mode, cl::sycl::access::target::global_buffer>
 		return *this;
 	}
 
-	template <cl::sycl::access::mode M = Mode, int D = Dims>
-	std::enable_if_t<detail::access::mode_traits::is_producer(M) && M != cl::sycl::access::mode::atomic && (D > 0), DataT&> operator[](
+	template <cl::sycl::access_mode M = Mode, int D = Dims>
+	std::enable_if_t<detail::access::mode_traits::is_producer(M) && M != cl::sycl::access_mode::atomic && (D > 0), DataT&> operator[](
 	    cl::sycl::id<Dims> index) const {
 		return sycl_accessor[index - backing_buffer_offset];
 	}
 
-	template <cl::sycl::access::mode M = Mode, int D = Dims>
+	template <cl::sycl::access_mode M = Mode, int D = Dims>
 	std::enable_if_t<detail::access::mode_traits::is_pure_consumer(M) && (D > 0), DataT> operator[](cl::sycl::id<Dims> index) const {
 		return sycl_accessor[index - backing_buffer_offset];
 	}
 
-	template <cl::sycl::access::mode M = Mode, int D = Dims>
-	std::enable_if_t<M == cl::sycl::access::mode::atomic && (D > 0), cl::sycl::atomic<DataT>> operator[](cl::sycl::id<Dims> index) const {
+	template <cl::sycl::access_mode M = Mode, int D = Dims>
+	std::enable_if_t<M == cl::sycl::access_mode::atomic && (D > 0), cl::sycl::atomic<DataT>> operator[](cl::sycl::id<Dims> index) const {
 		return sycl_accessor[index - backing_buffer_offset];
```

Comment:
This generates a deprecation warning in hipSYCL, try disabling `-Wdeprecated-declarations` on this line